CruCentralCoast/Resources/ResourcesVC.swift

```diff
@@ -89,3 +97,69 @@ extension ResourcesVC: UICollectionViewDelegateFlowLayout {
         return collectionView.frame.size
     }
 }
+
+extension ResourcesVC: ResourcePresentingDelegate {
+    func presentResource(of type: ResourceType, resource: Resource) {
+        switch type {
+        case .audio:
+            if let audioResource = resource as? AudioResource {
+                let vc = UIStoryboard(name: "Resources", bundle: nil).instantiateViewController(AudioResourceDetailVC.self)
+                vc.resource = audioResource
+                vc.audioResourceDelegate = self
+                self.show(vc, sender: self)
+            }
+        case .video:
+            if let videoResource = resource as? VideoResource {
+                let vc = UIStoryboard(name: "Resources", bundle: nil).instantiateViewController(VideoResourceDetailVC.self)
+                vc.resource = videoResource
+                self.show(vc, sender: self)
+            }
+        case .article:
+            if let articleResource = resource as? ArticleResource {
+                let vc = UIStoryboard(name: "Resources", bundle: nil).instantiateViewController(ArticleResourceDetailVC.self)
+                vc.resource = articleResource
+                self.show(vc, sender: self)
+            }
+        }
+    }
+}
+
+extension ResourcesVC: AudioResourceDelegate {
+    func playAudioFromURL(url: URL, title: String) {
+        self.audioController.playAudioFromURL(url: url)
+        self.audioController.titleLabel.text = title
+        self.revealMiniAudioPlayer()
+    }
+    
+    func dismissMiniAudioPlayer() {
+        UIView.animate(withDuration: 0.3) {
+            self.miniAudioPlayerTopConstraint?.constant = 0
+            self.view.layoutIfNeeded()
+        }
+    }
+    
+    func revealMiniAudioPlayer() {
+        UIView.animate(withDuration: 0.3) {
+            self.miniAudioPlayerTopConstraint?.constant = -self.audioController.bounds.height
+            self.view.layoutIfNeeded()
+        }
+    }
+    
+    func showAVPlayerViewController(player: AVPlayer) {
+        let vc = AVPlayerViewController()
+        vc.player = player
+        vc.addObserver(self.audioController, forKeyPath: #keyPath(AVPlayerViewController.view.frame), options: [], context: nil)
```

Comment:
What is this observer used for?

Reply:
I think i addressed this in one of my replies to one of your other comments. but if you're still confused let me know.